app/models/manageiq/providers/vmware/infra_manager.rb

```diff
@@ -249,6 +249,10 @@ def vm_relocate(vm, options = {})
       invoke_vim_ws(:relocateVM, vm, options[:user_event], options[:host], options[:pool], options[:datastore], options[:disk_move_type], options[:transform], options[:priority], options[:disk])
     end
 
+    def vm_move_into_folder(vm, options = {})
+      invoke_vim_ws(:moveIntoFolder, options[:folder], options[:user_event], vm.ems_ref_obj)
```

Comment:
Is this right?  `:moveIntoFolder` is defined on `MiqVimFolder` and takes just the managed object ems_ref_obj.  The method on `MiqVim` is `:moveIntoFolder_Task` and takes a folder ref and a vm ref.

Can you do a live test of this end to end to make sure it works?

Reply:
@agrare This has been end-to-end tested and it went well.
[```invoke_vim_ws```](https://github.com/ManageIQ/manageiq-providers-vmware/blob/d3f366695cc0e0480f40b70a1a806d85e328a3e6/app/models/manageiq/providers/vmware/infra_manager.rb#L447) would convert ```options[:folder]``` to ```MiqVimFolder``` and calls ```moveIntoFolder``` on it.

Reply:
Oh sorry you're completely right